Conventional/Conventions/Cecil/MustNotUsePropertyGetterSpecification.cs

```diff
@@ -2,39 +2,25 @@
 using System.Linq;
 using System.Linq.Expressions;
 using System.Reflection;
+using System.Runtime.CompilerServices;
 using Mono.Cecil;
 using Mono.Cecil.Cil;
 
 namespace Conventional.Conventions.Cecil
 {
     public abstract class MustNotUsePropertyGetterSpecification<TClass, TMember> : ConventionSpecification
     {
-        private class GetterDetails
-        {
-            public GetterDetails(string declaringType, string methodName)
-            {
-                DeclaringType = declaringType;
-                MethodName = methodName;
-            }
-
-            public string DeclaringType { get; private set; }
-            public string MethodName { get; private set; }
-        }
-
-        private readonly string _failureMessage;
-
-        private readonly GetterDetails[] _getterDetails;
+        private readonly (string DeclaringType, string MethodName)[] _getterDetails;
 
         protected MustNotUsePropertyGetterSpecification(Expression<Func<TClass, TMember>> expression,
             string failureMessage)
             :this(new[] {expression},failureMessage)
         {
-            
         }
 
         protected MustNotUsePropertyGetterSpecification(Expression<Func<TClass, TMember>>[] expressions, string failureMessage)
         {

```

Comment:
The expression parameter to this method is interesting - is it just a convenience?

My use case here is to disallow calls to a particular method, but this convention only supports getters. If the constructor supported accepting `MethodInfo`'s instead of `Expression`'s, then I wonder if this convention could apply to any method call instead? You then wouldn't need to find the property info of the expression body.

The `no Datetime.Now` convention would only change to be:

`typeof(DateTime).GetProperty("Now").GetGetMethod()` instead of `x=>DateTime.Now`

And the generic constraints would no longer be required. Maybe thats a new convention, not sure!

Reply:
Thinking further, given the complexity of this convention in handling generated async/iterator types, the more broadly applicable this convention is the better else others risk making these mistakes. Would love to not have to maintain my own convention for methods! 😊